src/main/java/com/github/redouane59/twitter/helpers/RequestHelper.java

```diff
@@ -49,6 +54,41 @@
     return Optional.ofNullable(result);
   }
 
+  public <T> Optional<T> uploadMedia(String url, File file, Class<T> classType) {
+    T result = null;
+    try {
+      OkHttpOAuthConsumer
+          consumer =
+          new OkHttpOAuthConsumer(TwitterClient.TWITTER_CREDENTIALS.getApiKey(), TwitterClient.TWITTER_CREDENTIALS.getApiSecretKey());
+      consumer.setTokenWithSecret(TwitterClient.TWITTER_CREDENTIALS.getAccessToken(), TwitterClient.TWITTER_CREDENTIALS.getAccessTokenSecret());
+
+      OkHttpClient client = new OkHttpClient.Builder()
```

Comment:
It's OkHttpClient anti-pattern:  creation of just one client per method call. Ideally it should be one per app. But at the same time using interceptors here is not convenient, because this signing will influence other places where the same OkHttpClient being used.
My recommendation: either take a look on ScribeJava, or use "dynamic" interceptors that bound to a particular request.

Reply:
I'll make a first PR to create a singleton because you're right that's really bad. I'll check after ScribeJava, it looks interesting, thank you very much for the suggestion !

Reply:
https://github.com/redouane59/twittered/pull/124